mux/src/tmux.rs

```diff
@@ -206,6 +329,32 @@ impl TmuxDomain {
 
 #[async_trait(?Send)]
 impl Domain for TmuxDomain {
+    async fn spawn(
+        &self,
+        _size: TerminalSize,
+        _command: Option<CommandBuilder>,
+        _command_dir: Option<String>,
+        _window: WindowId,
+    ) -> anyhow::Result<Arc<Tab>> {
+        self.inner.create_tmux_window();
+        anyhow::bail!("Intention: we use tmux command to do so");
+    }
+
+    async fn split_pane(
+        &self,
+        _source: SplitSource,
+        tab: TabId,
+        pane_id: PaneId,
+        split_request: SplitRequest,
+    ) -> anyhow::Result<Arc<dyn Pane>> {
+        self.inner.split_tmux_pane(tab, pane_id, split_request);
+
+        // Give a fake id for now, and fix it later on event WindowPaneChanged
```

Comment:
is there a risk that this fake id will cause the original pane to get broken or removed in the wrong circumstances later?

If so, then you could `alloc_pane_id()` a unique and unused id?